src/data_structures/fenwick.rs

```diff
@@ -1,28 +1,60 @@
+/// Fenwick Tree
+///
+/// Solves point add and range sum query
+///
+/// # Example
+/// ```
+/// use crate::programming_team_code_rust::data_structures::fenwick::Fenwick;
+///
+/// let mut fenwick = Fenwick::new(10);
+/// fenwick.add(0, 1);
+/// fenwick.add(1, 2);
+/// fenwick.add(2, 3);
+/// assert_eq!(fenwick.sum(0..3), 6);
+/// assert_eq!(fenwick.sum(0..2), 3);
+/// assert_eq!(fenwick.sum(1..3), 5);
+/// ```
 pub struct Fenwick<T> {
     ary: Vec<T>,
 }
 impl<T: Clone + Default + std::ops::AddAssign<T>> Fenwick<T> {
+    /// Creates a new Fenwick Tree with size n
+    ///
+    /// # Complexity
+    /// - Time: O(n)
+    /// - Space: O(n)
     pub fn new(n: usize) -> Self {
         Fenwick {
             ary: vec![T::default(); n],
         }
     }
-    pub fn accum(&self, mut idx: usize) -> T {
+
+    fn accum(&self, mut idx: usize) -> T {
         let mut sum = T::default();
         while idx > 0 {
             sum += self.ary[idx - 1].clone();
             idx &= idx - 1;
         }
         sum
     }
-    /// performs data[idx] += val;
+
+    /// Increments the value at `idx` by `val`
```

Comment:
clean; i like how minimal it is; Ya i guess here u don't need to document the params specifically